mesa/datacollection.py

```diff
@@ -197,17 +198,19 @@ def collect(self, model):
             for var, reporter in self.model_reporters.items():
                 # Check if lambda or partial function
                 if isinstance(reporter, (types.LambdaType, partial)):
-                    self.model_vars[var].append(reporter(model))
+                    self.model_vars[var].append(deepcopy(reporter(model)))
```

Comment:
Needs a brief explanatory comment before this line. Otherwise someone might mistakenly remove the deepcopy.

Reply:
Agreed, done.